src/System.Text.Json/tests/JsonNumberTests.cs

```diff
@@ -0,0 +1,651 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonNumberTests
+    {
+        private delegate bool TryGetValue<T>(JsonNumber number, out T result);
+
+        private static void TestInitialization<T>(
+                T value,
+                Func<T, JsonNumber> ctor,
+                Action<JsonNumber, T> setter,
+                Func<JsonNumber, T> getter,
+                TryGetValue<T> tryGetter,
+                Func<T, JsonNumber> implicitCaster)
+        {
+            // Default constructor:
+            JsonNumber number = new JsonNumber();
+            setter(number, value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // Numeric type constructor:
+            number = ctor(value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // String constructor:
+            number = value switch
+            {
+                double doubleValue => new JsonNumber(doubleValue.ToString(CultureInfo.InvariantCulture)),
+                float floatValue => new JsonNumber(floatValue.ToString(CultureInfo.InvariantCulture)),
+                decimal decimalValue => new JsonNumber(decimalValue.ToString(CultureInfo.InvariantCulture)),
```

Comment:
A bunch of tests are failing because we don't use InvariantCulture when converting the numbers to string within the implementation (in both SetDouble and GetDouble, i.e. double -> string and double -> string), but are expecting the result to be culture invariant for some reason.

Why not leave it as default (using `CultureInfo.CurrentCulture`)?

```C#
double val = 3.14;
CultureInfo.CurrentCulture = new CultureInfo( "es-ES", false );
Console.WriteLine("CurrentCulture is now {0}.", CultureInfo.CurrentCulture.Name);
// when culture changed to spanish, period became comma in the output
Console.WriteLine(val.ToString()); // 3,14
Console.WriteLine(val.ToString(CultureInfo.InvariantCulture)); // 3.14

// 3.14 - since both sides use same culture (parse and ToString), this round trips fine
Console.WriteLine(double.Parse(val.ToString()));

// 314 - ToString was invariant, but parsing wasn't - so can't roundtrip (culture mismatch)
Console.WriteLine(double.Parse(val.ToString(CultureInfo.InvariantCulture))); 

// 3.14 - if both sides match, this round trips fine
Console.WriteLine(double.Parse(val.ToString(CultureInfo.InvariantCulture), CultureInfo.InvariantCulture));
```

Reply:
> Why not leave it as default (using CultureInfo.CurrentCulture)?

Because it then can't be sent to Utf8JsonWriter as a formatted number?

Reply:
My question was assuming the implementation will remain as is.

> Because it then can't be sent to Utf8JsonWriter as a formatted number?

In that case, we should change the underlying representation to be culture invariant (in the impl itself).

Reply:
I changed the implementation to use Culture.Invariant.